src/Angor/Client/Pages/Browse.razor

```diff
@@ -320,13 +320,30 @@ else
         try
         {
             // Attempt to retrieve primary indexer and relay
+            var primaryExplorer = _NetworkService.GetPrimaryExplorer();
             var primaryIndexer = _NetworkService.GetPrimaryIndexer();
             var primaryRelay = _NetworkService.GetPrimaryRelay();
 
             // Initialize flags to track service availability
+            bool explorerAvailable = false;
             bool indexerAvailable = false;
             bool relayAvailable = false;
 
+            // Check if the primary explorer is available
+            if (primaryExplorer != null)
```

Comment:
do we actually need to check an explorer is online?

Reply:
I guess there is no harm in that

Reply:
It's not necessary, but showing the user a message that the explorer should also be included wouldn't be a bad idea.